src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs

```diff
@@ -205,7 +204,19 @@ private object FormatArgument(object? value)
             var enumerable = value as IEnumerable;
             if (enumerable != null)
             {
-                return string.Join(", ", enumerable.Cast<object>().Select(o => o ?? NullValue));
+                var vsb = new ValueStringBuilder(stackalloc char[256]);
+                bool first = true;
+                foreach (object e in enumerable)
+                {
+                    if (!first)
+                    {
+                        vsb.Append(", ");
+                    }
+
+                    vsb.Append(e != null ? e.ToString() : NullValue);
+                    first = false;
+                }
+                return vsb.ToString();
```

Comment:
This code has a different behaviour, the new one will put a comma after the last element while the old one won't.

Reply:
Actually, I've misread it, the comma is appended before the entry so everything is fine.